include/gatekeeper_gk.h

```diff
@@ -122,6 +140,13 @@ struct gk_fib {
 			 */
 			struct ether_cache eth_cache;
 		} grantor;
+
+		/*
+		 * When the action is GK_FWD_BACK_NET or GK_FWD_FRONT_NET,
+		 * we need the next fib entry for the gateway of the
+		 * destination.
+		 */
+		struct gk_fib *next_fib;
```

Comment:
One more note, once the perfect hash above is implemented, it will simplify the grantor entries, as already mentioned above, **and** the gateways entries since both entries will just point to the Ethernet headers at the neighbor entries.